File: backend/field-service/src/main/java/com/agroanalytics/field/model/Field.java
package com.agroanalytics.field.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "fields")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Field {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Double area;

    @Column(name = "crop_type")
    private String cropType;

    @Column(nullable = false)
    private String status;

    @Column(nullable = false)
    private Double lat;

    @Column(nullable = false)
    private Double lng;

    @Column(name = "soil_type")
    private String soilType;

    @Column(name = "planting_date")
    private LocalDate plantingDate;

    @Column(name = "expected_harvest_date")
    private LocalDate expectedHarvestDate;

    @Column(name = "current_moisture_level")
    private Double currentMoistureLevel;

    @Column(name = "organization_id")
    private Long organizationId;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/domain/entities/Field.ts
export type CropType =
  | 'wheat'
  | 'corn'
  | 'sunflower'
  | 'barley'
  | 'soy'
  | 'sugar_beet'
  | 'other'

export type FieldStatus = 'active' | 'idle' | 'harvested' | 'preparing'

export interface Coordinates {
  lat: number
  lng: number
}

export interface Field {
  id: string
  name: string
  area: number // hectares
  cropType: CropType
  status: FieldStatus
  coordinates: Coordinates
  polygon?: Coordinates[]
  soilType: string
  plantingDate?: string
  expectedHarvestDate?: string
  currentMoistureLevel?: number // %
  createdAt: string
  updatedAt: string
}

export interface CreateFieldDto {
  name: string
  area: number
  cropType: CropType
  status: FieldStatus
  coordinates: Coordinates
  soilType: string
  plantingDate?: string
  expectedHarvestDate?: string
}
